starters/next-drupal-starter/lib/getPreview.js

```diff
@@ -0,0 +1,41 @@
+import {
+  globalDrupalStateAuthStores,
+  getCurrentLocaleStore,
+} from "./drupalStateContext";
+import { fetchJsonapiEndpoint } from "@pantheon-systems/drupal-kit";
+/**
+ *
+ * @param {import('next').GetServerSidePropsContext |
+ * import('next').GetStaticPropsContext} previewData Nextjs context
+ * @param {string} node The node to be previewed example 'node--article'
+ */
+export async function getPreview(context, node) {
+  const store = getCurrentLocaleStore(
+    context.locale,
+    globalDrupalStateAuthStores
+  );
+  let requestInit = {};
+  if (process.env.CLIENT_ID && process.env.CLIENT_SECRET) {
+    requestInit = {
+      headers: {
+        Authorization: await store.getAuthHeader(),
+      },
+    };
+  }
+  const fetchedPreviewData = await fetchJsonapiEndpoint(
+    `${store.apiRoot}decoupled-preview/${context.previewData.key}?include=${store.params.getQueryObject().include}`,
```

Comment:
Previewing revisions is throwing a 500 for me. context.previewData.key is undefined for some reason.
<img width="980" alt="Screenshot at Jun 01 09-31-39" src="https://user-images.githubusercontent.com/889478/171430235-6c893a3f-ee94-4ccc-ab7a-02b74bcafb61.png">

Reply:
Even after adding the key, I am unable to fetch the data from the revision

When visiting the fetched link directly I see the following:

```
The website encountered an unexpected error. Please try again later.
```

Is there something special we need to do from the frontend for revisions?